src/main/scala/com/advancedtelematic/director/http/AssignmentsResource.scala

```diff
@@ -51,24 +55,33 @@ class AssignmentsResource(extractNamespace: Directive1[Namespace])
             val f = createAssignments(ns, req).map(_ => StatusCodes.Created)
             complete(f)
           }
+        } ~
+        patch {
+          entity(as[CancelAssignments]) { cancelAssignments =>
+            val a = deviceAssignments.cancel(ns, cancelAssignments.cancelAssignments).flatMap { canceledAssignments =>
+              Future.traverse(canceledAssignments) { canceledAssignment =>
+                val ev: DeviceUpdateEvent = DeviceUpdateCanceled(ns, Instant.now, canceledAssignment.correlationId, canceledAssignment.deviceId)
+                messageBusPublisher.publish(ev).map(_ => canceledAssignment)
+              }
+            }
+            complete(a.map(_.map(_.deviceId)))
+          }
         }
-        // TODO:SM
-//        patch {
-//          // Seq[DeviceId
-//          entity(as[String]) { _ =>
-//            ???
-//          }
-//        }
       } ~
       pathPrefix(DeviceId.Path) { deviceId =>
         get { //  This should be replacing /queue in /admin
           val f = deviceAssignments.findDeviceAssignments(ns, deviceId)
           complete(f)
+        } ~
+        delete {
```

Comment:
Where is this coming from? What is it for?

Reply:
Yeah I meant the endpoint. Why do we need PATCH and DELETE ? Could we just support one of them?

Reply:
Yeah I don't know. Maybe the DELETE one was first? Doesn't make sense to keep it now, true.